docs/deployment/nativeaot.md

```diff
@@ -0,0 +1,141 @@
+
+When Native AOT deployment is enabled, the build system analyzes your code, and all its dependencies, to verify if it's suitable for full trimming and AOT compilation. If incompatibilities are detected, trimming and AOT warnings are produced. A single trimming or AOT warning means that the app isn't compatible with Native AOT deployment, and that it might not work correctly. Therefore, when building an app for Native AOT deployment you should review and correct all trimming and AOT warnings. Failure to do this may result in exceptions at runtime since necessary code could have been removed. If you suppress the warnings the AOT deployed app must be thoroughly tested to verify that functionality hasn't changed from the untrimmed app. For more information, see [Introduction to trim warnings](/dotnet/core/deploying/trimming/fixing-warnings) and [Introduction to AOT warnings](/dotnet/core/deploying/native-aot/fixing-warnings).
+
+> [!NOTE]
+> There may be cases where fixing trimming and AOT warnings isn't possible, such as when they occur for third-party libraries. In such cases, third-party libraries will need to be updated to become fully compatible.
+
+## Native AOT performance benefits
+
+Publishing and deployment a Native AOT app produces an app that's typically up to 2.5x smaller, and an app that starts up typically up to 2x faster.
+
+The following chart shows the resulting app package size for a `dotnet new maui` app on iOS and Mac Catalyst across different deployment models:
+
+:::image type="content" source="media/nativeaot/app-package-size.png" alt-text="Chart showing app package size across different deployment models." border="false":::
+
+The preceding chart shows that Native AOT produces more than 2x smaller apps for both iOS and Mac Catalyst compared to the default deployment model.
+
+The following chart shows the average startup time for a `dotnet new maui` app on iOS and Mac Catalyst on Mono and Native AOT deployment:
+
+:::image type="content" source="media/nativeaot/average-startup-time.png" alt-text="Chart showing average app startup time on Mono and Native AOT." border="false":::
+
+The preceding chart shows that Native AOT has up to 2x faster startup times on iOS devices and 1.2x faster startup time on Mac Catalyst, compared to Mono deployment.
+
+The following chart shows the average build time for a `dotnet new maui` app on iOS and Mac Catalyst across different deployment models:
+
+:::image type="content" source="media/nativeaot/average-build-time.png" alt-text="Chart showing average app build time on Mono and Native AOT." border="false":::
+
+The preceding chart shows that Native AOT has up to 2.8x faster build times on iOS devices compared to the default deployment model. For Mac Catalyst, build times are comparable for arm64 single RID apps, but are slightly slower for universal apps when compared to Mono deployment.
```

Comment:
I am not sure if these graph should end up in the official documentation (or in this form). They were measured with RC1 release and numbers will most probably change if we compare them against GA (and future servicing releases). So with that, would be good to avoid someone taking them for granted. 

Maybe we could add a sentence which would clarify this if we decide to keep the graphs/numbers in our official docs, something like: "Our preliminary .NET 9 performance testing is showing the following results... which are presented here to give an idea of achievable performance improvements with Native AOT deployment model"

@MichalStrehovsky @simonrozsival what do you think?

Reply:
I don't have a strong opinion either way. ASP.NET does have similar graph: https://learn.microsoft.com/en-us/aspnet/core/fundamentals/native-aot?view=aspnetcore-8.0#why-use-native-aot-with-aspnet-core. It will get stale, but the ballpark will probably be the same for a long time.

Reply:
Thanks! @davidbritch already added a note about it here: https://github.com/dotnet/docs-maui/pull/2579/files#diff-3d87ee068ce7c17f2f37fffcaf5bae0c6654b39221f7673172d8c55d3aba2dd8R30

So I think we are good to go.